pkg/agent/manager/private.go

```diff
@@ -0,0 +1,115 @@
+package manager
+
+import (
+	"crypto/ecdsa"
+	"crypto/x509"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/spiffe/spire/pkg/agent/manager/cache"
+	proto "github.com/spiffe/spire/proto/common"
+
+	tomb "gopkg.in/tomb.v2"
+)
+
+type manager struct {
+	c     *Config
+	t     *tomb.Tomb
+	cache cache.Cache
+
+	stopped chan error
+
+	// Fields protected by mtx mutex.
+	mtx     *sync.RWMutex
+	svid    *x509.Certificate
+	svidKey *ecdsa.PrivateKey
+	bundle  []*x509.Certificate // Latest CA bundle
+
+	spiffeID       string
+	serverSPIFFEID string
+	serverAddr     *net.TCPAddr
+
+	svidCachePath   string
+	bundleCachePath string
+
+	clients *clientsPool
+}
+
+func (m *manager) run() error {
+	m.t.Go(m.synchronizer)
+	m.t.Go(m.rotator)
+	return nil
+}
+
+func (m *manager) synchronizer() error {
+	t := time.NewTicker(5 * time.Second)
+	for {
+		select {
+		case <-t.C:
+			err := m.synchronize()
+			if err != nil {
+				return err
```

Comment:
does this mean manager will fail/stop if we encounter an error during synchronization? same question for rotator below

Reply:
Yes, this would make the manager stop with an error (same for rotator). We should decide if we want to restart the manager or just log the synchronize error and keep the synchronizer running. Which do you think would be the best approach?